File: data/data.code.golang.client/req_test_js_ok.go
package proto

import (
	"packet"
)

type ReqTestJsOk struct {
	u64                      uint64
	i64                      int64
}

func (this *ReqTestJsOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint64(this.u64)
	pack.WriteInt64(this.i64)

	return pack.Encode(P_REQ_TEST_JS_OK)
}

func (this *ReqTestJsOk) SetU64(u64 uint64) {
	this.u64 = u64
}

func (this *ReqTestJsOk) SetI64(i64 int64) {
	this.i64 = i64
}

File: data/data.code.lua.client/ReqTestJsOk.lua
module("ReqTestJsOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqTestJsOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.u64 = 0
	self.i64 = 0
end


function encode(self)
	pack = packet.new()
	pack:writeu64(self.u64)
	pack:writei64(self.i64)
	return pack:encode(Msg.P_REQ_TEST_JS_OK)
end


function setU64(self, u64)
	self.u64 = u64
end

function setI64(self, i64)
	self.i64 = i64
end

